### PR TITLE
Fix populate of array types

### PR DESCRIPTION
**Pull request by kjkurtz**

### CODE REVIEW

**Review by joshgachnang — APPROVED** [review]
Nice, makes sense! Good catch.

**Review by joshgachnang — CHANGES_REQUESTED** [review]
Can you add this to tests? Like making sure "eatenBy.userId" works on the FoodModel?

File: src/__snapshots__/openApi.test.ts.snap [hunk 1]
```diff
@@ -103,6 +103,52 @@ Object {
                             },
                             "type": "array",
                           },
+                          "eatenBy.userId": Object {
```
Comment (joshgachnang) [inline src/__snapshots__/openApi.test.ts.snap:106]: It looks like these came out as objects instead of an array?

Comment (joshgachnang) [reply src/__snapshots__/openApi.test.ts.snap:106]: Though if it generates correctly in the SDK, I'm ok with it.

Comment (kjkurtz) [reply src/__snapshots__/openApi.test.ts.snap]: It'll be an object but have an `items` list.

Comment (kjkurtz) [reply src/__snapshots__/openApi.test.ts.snap]: Screenshot 2023-08-22 at 10 06 04 PM
for example, this is what is generated for notifications.

**Review by joshgachnang — APPROVED** [review]
Sweet!

**Pull request merged**